conductr_cli/docker.py

```diff
@@ -48,3 +50,106 @@ def cpu_check(info):
     existing_cpu_count = int(m.group(1))
     has_sufficient_cpu = existing_cpu_count >= DEFAULT_DOCKER_CPU_COUNT
     return existing_cpu_count, has_sufficient_cpu
+
+
+@validation.handle_vbox_manage_not_found_error
+def validate_docker_vm(vm_type):
+    log = logging.getLogger(__name__)
+    if vm_type is DockerVmType.DOCKER_ENGINE:
+        try:
+            info = terminal.docker_info()
+            existing_ram, has_sufficient_ram = ram_check(info)
+            existing_cpu, has_sufficient_cpu = cpu_check(info)
+
+            if not has_sufficient_ram or not has_sufficient_cpu:
+                if not has_sufficient_ram:
+                    log.warning('Docker has insufficient RAM of {} GiB - please increase to a minimum of {} GiB'
+                                .format(existing_ram, DEFAULT_DOCKER_RAM_SIZE))
+
+                if not has_sufficient_cpu:
+                    log.warning('Docker has an insufficient no. of CPUs {} - please increase to a minimum of {} CPUs'
+                                .format(existing_cpu, DEFAULT_DOCKER_CPU_COUNT))
+        except (AttributeError, CalledProcessError):
+            log.error('Docker native is installed but not running.')
+            log.error('Please start Docker with one of the Docker flavors based on your OS:')
+            log.error('  Linux:   Docker service')
+            log.error('  MacOS:   Docker for Mac')
+            log.error('  Windows: Docker for Windows')
+            log.error('A successful Docker startup can be verified with: docker info')
+            exit(1)
```

Comment:
After further discussion, we decided not to pursue this. The HAProxy stop can trap docker related error and then continue silently.